core/templates/dev/head/pages/email-dashboard-pages/email-dashboard-data.service.ts

```diff
@@ -16,109 +16,121 @@
  * @fileoverview Services for oppia email dashboard page.
  */
 
-angular.module('oppia').factory('EmailDashboardDataService', [
-  '$http', '$q', function($http, $q) {
-    var QUERY_DATA_URL = '/emaildashboarddatahandler';
-    var QUERY_STATUS_CHECK_URL = '/querystatuscheck';
-    // No. of query results to display on a single page.
-    var QUERIES_PER_PAGE = 10;
-    // Store latest cursor value for fetching next query page.
-    var latestCursor = null;
-    // Array containing all fetched queries.
-    var queries = [];
-    // Index of currently-shown page of query results.
-    var currentPageIndex = -1;
-
-    var fetchQueriesPage = function(pageSize, cursor) {
-      return $http.get(QUERY_DATA_URL, {
-        params: {
-          num_queries_to_fetch: pageSize,
-          cursor: cursor
-        }
-      }).then(function(response) {
-        return response.data;
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+export interface Query {
+  id: string,
+  status: string
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class EmailDashboardDataService {
+  QUERY_DATA_URL: string = '/emaildashboarddatahandler';
+  QUERY_STATUS_CHECK_URL: string = '/querystatuscheck';
+  // No. of query results to display on a single page.
+  QUERIES_PER_PAGE: number = 10;
+  // Store latest cursor value for fetching next query page.
+  latestCursor = null;
+  // Array containing all fetched queries.
+  queries: Array<Query> = [];
+  // Index of currently-shown page of query results.
+  currentPageIndex: number = -1;
+
+  constructor(
+    private http: HttpClient
+  ) {}
+
+  fetchQueriesPage(pageSize, cursor): Promise<Object> {
+    return this.http.get(this.QUERY_DATA_URL, {
+      params: {
+        num_queries_to_fetch: pageSize,
+        cursor: cursor
+      }
+    }).toPromise();
+  }
+
+  getQueries(): Array<Query> {
+    return this.queries;
+  }
+
+  getCurrentPageIndex(): number {
+    return this.currentPageIndex;
+  }
+
+  getLatestCursor() {
```

Comment:
Missing return type

Reply:
I'm not sure about its return type as `latestCursor` is used everywhere as `null`.